### PR TITLE
Return the correct errors to users when an mfa ceremony fails

### PR DESCRIPTION
**Pull request by rosstimothy**
Errors returned from the mfa prompt are now wrapped to allow them to be differentiated and acted upon accordingly by callers. No further processing of the error is required in this change, the code that examines the errors will already do the right thing.

Fixes #35580.

changelog: Prevent tsh from re-authenticating if the mfa ceremony fails during tsh ssh

### CODE REVIEW

**Comment by flyinghermit** [comment]
Can we only print the `User Message` by default and print `ERROR REPORT...Original Error...Stack Trace:` when in debug mode?

File: lib/client/cluster_client.go [hunk 1]
```diff
@@ -84,6 +84,20 @@ func (c *ClusterClient) Close() error {
 	return trace.NewAggregate(c.AuthClient.Close(), c.ProxyClient.Close())
 }
 
+// ceremonyFailedErr indicates that the mfa ceremony was attempted unsuccessfully.
```
Comment (flyinghermit) [inline lib/client/cluster_client.go:87]:
```suggestion
// ceremonyFailedErr indicates failed mfa ceremony.
```
is that the same thing? ignore otherwise.

**Review by flyinghermit — APPROVED** [review]
> Ah the output that I included above was from tsh -d. I'll update to show output from with/without the debug flag.

sorry I thought that's the default behavior.

File: lib/client/cluster_client.go [hunk 2]
```diff
@@ -151,7 +165,7 @@ func (c *ClusterClient) SessionSSHConfig(ctx context.Context, user string, targe
 	log.Debug("Issued single-use user certificate after an MFA check.")
 	am, err := key.AsAuthMethod()
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, trace.Wrap(ceremonyFailedErr{err})
```
Comment (GavinFrazar) [inline lib/client/cluster_client.go:168]: why wrap it here?

And what underlying error type are we hiding that causes the issue?

**Review by GavinFrazar — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Pull request merged**